File: app/src/main/java/com/cv4j/app/activity/DetectQRActivity.java
/*
 * Copyright (c) 2017 - present, CV4J Contributors.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.cv4j.app.activity;

import android.os.Bundle;
import android.support.design.widget.TabLayout;
import android.support.v4.app.Fragment;
import android.support.v4.view.ViewPager;
import android.support.v7.widget.Toolbar;

import com.cv4j.app.R;
import com.cv4j.app.adapter.QRAdapter;
import com.safframework.injectview.annotations.InjectExtra;
import com.safframework.injectview.annotations.InjectView;
import com.safframework.injectview.annotations.OnClick;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by Tony Shen on 2017/6/25.
 */

public class DetectQRActivity extends BaseActivity {

    @InjectView(R.id.tablayout)
    TabLayout mTabLayout;

    @InjectView(R.id.viewpager)
    ViewPager mViewPager;

    @InjectView(R.id.toolbar)
    Toolbar toolbar;

    @InjectExtra(key = "Title")
    String title;

    private List<Fragment> mList = new ArrayList();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_detect_qr);

        initData();
    }

    private void initData() {
        toolbar.setTitle("< "+title);

        for(int i=0;i<4;i++) {
            mList.add(QRFragment.newInstance(i));
        }

        mViewPager.setAdapter(new QRAdapter(this.getSupportFragmentManager(),mList));
        mTabLayout.setupWithViewPager(mViewPager);
    }

    @OnClick(id= R.id.toolbar)
    void clickToolbar() {

        finish();
    }
}

File: app/src/main/java/com/cv4j/app/activity/DetectQRActivity.kt
package com.cv4j.app.activity

import android.os.Bundle
import androidx.fragment.app.Fragment
import com.cv4j.app.R
import com.cv4j.app.adapter.QRAdapter
import com.cv4j.app.app.BaseActivity
import com.cv4j.app.fragment.QRFragment
import kotlinx.android.synthetic.main.activity_detect_qr.*
import java.util.*

/**
 *
 * @FileName:
 *          com.cv4j.app.activity.DetectQRActivity
 * @author: Tony Shen
 * @date: 2020-05-05 16:13
 * @version: V1.0 <描述当前版本功能>
 */

class DetectQRActivity : BaseActivity() {

    var title: String? = null

    private val mList: MutableList<Fragment> = ArrayList<Fragment>()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_detect_qr)

        intent.extras?.let {
            title = it.getString("Title","")
        }?:{
            title = ""
        }()

        toolbar.setOnClickListener {
            finish()
        }

        initData()
    }

    private fun initData() {
        toolbar.title = "< $title"
        for (i in 0..3) {
            mList.add(QRFragment.newInstance(i))
        }
        viewpager.adapter = QRAdapter(this.supportFragmentManager, mList)
        tablayout.setupWithViewPager(viewpager)
    }
}